doc/nrf/security/crypto/crypto_architecture.rst

```diff
@@ -0,0 +1,143 @@
+5. The implementation delegates the actual cryptographic processing to the selected crypto driver.
+   The driver performs the requested operation using either hardware acceleration or software algorithms.
+6. The driver returns the operation result to the PSA Crypto implementation, which performs any necessary post-processing and validation.
+7. The implementation returns the final result to the application through the PSA Crypto API, completing the cryptographic operation.
+
+.. _ug_crypto_architecture_implementation_standards:
+
+Implementations in the |NCS|
+****************************
+
+The |NCS| follows the PSA Crypto standard through two major API implementations: Oberon PSA Crypto and TF-M Crypto Service.
+They are designed for use without and with Trusted Firmware-M (TF-M), respectively, and have different security requirements.
+Both are based on the `sdk-oberon-psa-crypto`_ library, which offers a lightweight PSA Crypto API implementation optimized for resource-constrained microcontrollers.
+
+.. figure:: ../images/psa_crypto_api_overview.svg
+   :alt: PSA Crypto API implementations in the |NCS|
+   :align: center
+
+   PSA Crypto API implementations in the |NCS|
+
+.. _ug_crypto_architecture_implementation_standards_oberon:
+
+Oberon PSA Crypto implementation
+================================
+
+The Oberon PSA Crypto implementation provides a direct PSA Crypto API interface for applications that do not require Trusted Firmware-M (TF-M).
+This implementation is built around the Oberon PSA Crypto, a library that serves as the central component managing cryptographic operations and driver selection.
+
+.. figure:: ../images/psa_crypto_api_oberon.svg
+   :alt: Oberon PSA Crypto implementation
+   :align: center
+
+   Oberon PSA Crypto implementation
+
+The Oberon PSA Crypto acts as the implementation provider, directly exposing the PSA Crypto API to applications.
+Each driver can implement support for different subsets of cryptographic algorithms, providing software support for algorithms that hardware cannot support.
+
+The following figure shows the driver library selection through the driver wrapper, one of the internal modules of Oberon PSA Crypto:
+
+.. figure:: ../images/psa_certified_api_lib_selection.svg
+   :alt: Oberon PSA Crypto driver library selection
+   :align: center
+
+   Oberon PSA Crypto driver library selection
+
+This implementation standard is suitable for applications that prioritize simplicity and do not require the additional security isolation provided by TF-M.
+It offers direct access to cryptographic functionality with minimal overhead, making it ideal for resource-constrained applications.
+
+Storage integration for the Oberon PSA Crypto implementation
+------------------------------------------------------------
+
+When using the Oberon PSA Crypto implementation, persistent keys from the PSA Crypto API are stored in the |NCS| using one of the following storage mechanisms:
```

Comment:
Change "are stored" to "can be stored"?

Additionally, on the nRF54L series we also support storing persistent keys on the device's KMU. This is handled by the cracen driver, so this page might not be the best place for detailed explanations. However, we shouldn't claim the secure/trusted storage are the only integrated persistent storage locations.